File: phpopencloud/challenge-3.php
<?php

require 'vendor/autoload.php';
use OpenCloud\OpenStack;

#1: Connecting to the cloud

$cloud = new OpenStack('https://YOUR_IDENTITY_CONTROLLER', array(
    'username'   => 'YOUR_USERNAME',
    'password'   => 'YOUR_PASSWORD',
    'tenantName' => 'YOUR_PROJECT'
));

$compute = $cloud->computeService('nova', 'YOUR_REGION');

#3: Listing flavors 

echo ("**Available flavors** \n");
$flavors = $compute->flavorList();

foreach ($flavors as $flavor) {
     printf("ID: %s, Name: %s, RAM: %s, VCPUs: %s\n", $flavor->id, $flavor->name, $flavor->ram, $flavor->vcpus);
}


?>

File: fog/challenge-3.rb
require "fog/openstack"

#1: Connecting to the cloud

@connection_params = {
	openstack_auth_url:     "https://YOUR_IDENTITY_CONTROLLER",
	openstack_username:     "YOUR_USERNAME",
  	openstack_api_key:      "YOUR_PASSWORD",
  	openstack_project_name: "YOUR_PROJECT",
  	openstack_region: 	"YOUR_REGION", 
}

compute = Fog::Compute::OpenStack.new(@connection_params)

#3: Listing flavors 
p compute.flavors